src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java

```diff
@@ -12,26 +14,54 @@
 
 @Service
 public class BookServiceImpl implements BookService {
+    @Autowired
     private BookMapper bookMapper;
     @Autowired
     private BookRepository bookRepository;
 
     @Override
-    public BookDto save(CreateBookRequestDto createBookRequestDto) {
-        return bookMapper.bookToBookDto(bookRepository.save(
-                bookMapper.createBookRequestDtoToBook(createBookRequestDto)));
+    public BookDto save(BookRequestDto bookRequestDto) {
+        return bookMapper.toDtoBook(bookRepository.save(
+                bookMapper.toBook(bookRequestDto)));
     }
 
     @Override
     public List<BookDto> findAll() {
         return bookRepository.findAll().stream()
-                .map(bookMapper::bookToBookDto)
+                .map(bookMapper::toDtoBook)
                 .toList();
     }
 
     public BookDto getBookById(Long id) {
         return bookRepository.findById(id)
-                .map(bookMapper::bookToBookDto)
-                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+                .map(bookMapper::toDtoBook)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book not found with id: " + id));
+    }
+
+    @Override
+    public BookDto updateBook(BookUpdateDto bookUpdateDto, Long id) {
+        Book existingBook = bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book not found with id: " + id));
```

Comment:
you can do this check without creating existingBook variable. If not bookRepository.findById(id) throw exception

Reply:
> you can do this check without creating existingBook variable. If not bookRepository.findById(id) throw exception

But at this line i copy all not null values from bookUpdateDto to existingBook; And then save book.
bookMapper.updateBookFromDto(bookUpdateDto, existingBook);